check_diff/src/lib.rs

```diff
@@ -70,6 +76,21 @@ pub struct RustfmtRunner {
     binary_path: PathBuf,
 }
 
+impl CheckDiffRunners {
+    /// Creates a diff generated by running the source and feature binaries on the same file path
+    pub fn create_diff(
+        &self,
+        path: &Path,
+        additional_configs: &Option<Vec<String>>,
+    ) -> Result<String, CheckDiffError> {
+        let code = std::fs::read_to_string(path)?;
+        let src_format = self.src_runner.format_code(&code, additional_configs)?;
+        let feature_format = self.feature_runner.format_code(&code, additional_configs)?;
+        let diff = diffy::create_patch(src_format.as_str(), feature_format.as_str());
+        Ok(format!("{diff}"))
```

Comment:
I know this was in my suggestion when messaging on Zulip. I think this should be fine, but we could also consider returning a [`diffy::Patch`](https://docs.rs/diffy/latest/diffy/struct.Patch.html) or maybe even our own wrapper type.

It would be really nice if the `diff::Patch` gave us some way to tell if the patch was empty, but I don't think it does so we probably need to roll our own way to figure that out.

Reply:
To tell us whether there are any diffs, `diffy` uses [hunks](https://docs.rs/diffy/latest/diffy/struct.Hunk.html), we can simply check if there are any hunks to indicate the number of different lines.